first_ruby/db/seeds.rb

```diff
@@ -0,0 +1,7 @@
+# This file should contain all the record creation needed to seed the database with its default values.
```

Comment:
1. Whats this file?
2. How do we use this file?
3. Why do we need this file?

Reply:
1. This file is for writing database schema & migration rules.
2. By using `rake` command
3. To populate default data easily

Reply:
(1) wrong
(2) give the exact command
(3) define `default`

Reply:
1. To define the key and translation for the locale, it can also display the localize date and the defined format. 
2. In this case, the configured locale is `en` and the setting is on key `hello`
```
en:
    hello: "Hello world!"
```
To use this, in the `view` use the format `<%= t('hello') %>` and this will display `Hello world!` at the page when display. The `t` here stands for translation.

Reply:
i think you answer the above in the wrong comment

Reply:
Opps, my bad

Reply:
(1) This file is to create data easily for the database.

(2) To use this file, there will need a `model` class for creation and using the `.create` after the class model name and state the variable of the model along with the data. 
For example, the commented code here: 
`Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])`
Setup database first using `rake db:setup` if it isn't setup yet and `rake db:migrate` to create the table according to `Movie` model. After that, use `rake db:seed` to create the data.

(3) `default` is which to create sample data to use. For example the page will preview all `Movie` in the main page, since to verify the following nature, some sample data are required.

Reply:
(1) ok
(2) What if no model class? Can I not use this file to add data?
(3) do you mean need to create data for every model that you have?

Reply:
2. Since model is correspond to the database's structure, model is required.
3. I mean to generate some sample data to for development phase. As here let's say there's a page that would displays some movies title so I'll generate some `Movie `data to see if it's working to display it out.

Reply:
(2) 100% incorrect. 
(3) Ok

This is just a ruby file. Running `rails db:seeds` will invoke this ruby file with the models defined in the rails app. You don't need to seed any model or all model. You can run a normal ruby code to seed say an external database like redis that doesn't relate to any model here at all.

*Click "resolve conversation" once you faham & have read this

Reply:
I see, ok understood